network/transport/v2/handlers.go

```diff
@@ -133,24 +134,30 @@ func (p *protocol) handleTransactionPayloadQuery(ctx context.Context, peer trans
 		WithField(core.LogFieldTransactionRef, hash.FromSlice(msg.TransactionRef)).
 		Trace("Handling TransactionPayloadQuery")
 
+	// get peer's connection, authentication status is checked later on
+	connection := p.connectionList.Get(grpc.ByConnected(), grpc.ByNodeDID(peer.NodeDID), grpc.ByPeerID(peer.ID))
```

Comment:
does this also work for unauthenticated connections? (bootstrapping?)

Reply:
yes, but public payloads are attached to their transactions so this is for private TXs only.
#1940 changes this to respond on the connection the query came from.